src/app/api/stories/route.ts

```diff
@@ -126,52 +133,108 @@ export async function GET(req: Request) {
   }
 }
 
-export async function POST(request: NextRequest) {
-  const timestamp = new Date();
-  const parsedBody = postStorySchema.safeParse(await request.json());
-  if (!parsedBody.success) {
-    return NextResponse.json({ error: "Bad Request" }, { status: 400 });
+async function processThumbnailImage(
+  data: z.infer<typeof putStorySchema>,
+): Promise<string | null> {
+  if (data.image) {
+    const imageMimeType = data.image.type;
+    const extension =
+      imageMimeType === "image/jpeg"
+        ? "jpg"
+        : imageMimeType === "image/png"
+        ? "png"
+        : "gif";
+    const thumbnailFilename = `${randomUUID()}.${extension}`;
+    const thumbnailUrl = `${bucketUrlPrefix}${thumbnailFilename}`;
+    await bucket
+      .file(thumbnailFilename)
+      .save(Buffer.from(await data.image.arrayBuffer()));
+    return thumbnailUrl;
   }
+  return data.imageUrl ?? null;
+}
 
-  const { title, content } = parsedBody.data;
+export async function PUT(request: NextRequest) {
+  try {
+    const session = await getServerSession();
+    const user = await prisma.user.findUnique({
+      where: { email: session?.user.email ?? "noemail" },
+    });
 
-  const user = await prisma.user.findFirst({});
-  if (!user) {
-    return NextResponse.json({ error: "No user available" }, { status: 500 });
-  }
+    if (!user || !user.roles.includes("EDITOR")) {
```

Comment:
another small question I have that doesn't need to be answered right now: do all editors have the ability to edit all stories?  For example, if someone writes one story and becomes an editor, can they now edit another person's story?  I may have missed where we were checking that.